lib/tests.py

```diff
@@ -73,8 +84,26 @@ def test_route_for(plugin):
 
 def test_route_for_args(plugin):
     f = lambda: None
-    plugin.route("/foo/<a>/<b>")(f)
-    assert plugin.route_for(plugin.base_url + "/foo/1/2") is f
+    g = lambda: (None, None)  # just to make sure that they are easily different
+    plugin.route("/foo/<a>/<b2>")(f)
+    plugin.route("/foo/a/b")(g)
+
+    # due to the unpredictable sorting of dict, just do it 100 times to see if it fails
+    for i in range(0, 100):
+        assert plugin.route_for(plugin.base_url + "/foo/1/2") is f
+        assert plugin.route_for(plugin.base_url + "/foo/a/b") is g
+
+
+def test_path_args(plugin):
+    url = 'http://foo.bar:80/baz/bax.json?foo=bar&baz=bay'
+
+    def test_path_args_inner(something):
+        assert something == url
+
+    plugin.route('/do/<path:something>')(test_path_args_inner)
+    path = plugin.url_for(test_path_args_inner, url)
+    assert plugin.route_for(path) is test_path_args_inner
+    plugin.run([path, '0', ''])
```

Comment:
This should be verified with `f.assert_called_with(..)` instead of `assert` inside the target function. Otherwise the test wont fail if the router don't call the function property. You can just use the `f = lambda: None` pattern from the other tests.

Reply:
Oh. Admittedly, I didn't know you could do that. I've not enough experience with unit testing, let alone Python unit testing. So far, I've been happy that it's so painless in Python.